packages/react-components/react-nav-preview/src/components/Nav/useNav.ts

```diff
@@ -83,5 +92,62 @@ export const useNav_unstable = (props: NavProps, ref: React.Ref<HTMLDivElement>)
     onUnregister,
     onSelect,
     getRegisteredNavItems,
+    onRequestNavCategoryItemToggle,
+    openItems,
   };
 };
+
+// /**
+//  * Initial value for the uncontrolled case of the list of open indexes
+//  */
+// function initializeUncontrolledOpenItems({ defaultOpenItems }: Pick<NavProps, 'defaultOpenItems'>): NavItemValue[] {
+//   if (defaultOpenItems !== undefined) {
+//     if (Array.isArray(defaultOpenItems)) {
+//       return [defaultOpenItems[0]];
+//     }
+//     return [defaultOpenItems];
+//   }
+//   return [];
+// }
+
+// /**
+//  * Normalizes Accordion index into an array of indexes
+//  */
+// function normalizeValues(index?: NavItemValue | NavItemValue[]): NavItemValue[] | undefined {
+//   if (index === undefined) {
+//     return undefined;
+//   }
+//   return Array.isArray(index) ? index : [index];
+// }
+
+// temp implementation of the above function.
+function normalizeValues(index?: NavItemValue | NavItemValue[]): NavItemValue[] | undefined {
+  return undefined;
+}
+
+/**
+ * Updates the list of open indexes based on an index that changes
+ * @param value - the index that will change
+ * @param previousOpenItems - list of current open indexes
+ * @param multiple - if Nav support multiple SubItemGroups opened at the same time
+ * @param collapsible - if Nav support multiple SubItemGroups closed at the same time
+ */
+function updateOpenItems(
+  value: NavItemValue,
+  previousOpenItems: NavItemValue[],
+  multiple: boolean,
+  collapsible: boolean,
+) {
+  if (multiple) {
+    if (previousOpenItems.includes(value)) {
```

Comment:
Is this check needed? this adds another check for all the items in the list of values, but down in the filter in line 144 you are also going through the items. You could use a variable instead and go from two runs into one:
```jsx
let nextOpenItems: NavItemValue;
if(previousOpenItems.length > 1 || collapsible) {
  nextOpenItems = previousOpenItems.filter(i => i !== value);
}

return nextOpenItems?.length > 0 ? nextOpenItems : [...previousOpenItems, value].sort();
```

this is kind of a nit, but if the nav is going to contain many items, it will benefit a bit.

Reply:
Summary of offline convo - going to strip this function down to essentials and build back up as we add relevant props.